### ISSUE TITLE
ADS-B no map display on latest Docker build

### ISSUE THREAD

**Issue by spsx** [f4exb/sdrangel-docker#28]
I'm running sdrangel-docker, having built the latest 6.2.0 version today. The build completed with no issues. I use Ubuntu 18.04, hence why I use the Docker builds instead of the public releases.

The program runs great, however there is an issue with the new ADS-B feature. Fight data is collected fine and shows up in the "ADS-B Data" box, but the Map box is blank. The aircraft and airport databases have been downloaded and logs confirm they're loaded properly. See the screenshot below (note that there were no planes when I made the screenshot, but the map is still blank even when planes exist):

![Screenshot from 2020-11-29 21-08-48](https://user-images.githubusercontent.com/19350831/100561637-69af4500-3287-11eb-8497-c66ad4c06f14.png)

Another user [reported a similar issue](https://github.com/f4exb/sdrangel/issues/677), and looking at the replies I think that this occurs because the version of QT used in the Docker image is too old. Specifically [map.qml](https://github.com/f4exb/sdrangel/blob/master/plugins/channelrx/demodadsb/map/map.qml) looks for QT 5.12 and Quick 2.12, but the Docker image (Debian 10) uses QT 5.11 and Quick 2.11.

I have created a log of this issue, where I open sdrangel and start my ADS-B preset. I've attached it below:
[sdrangel.log](https://github.com/f4exb/sdrangel-docker/files/5613460/sdrangel.log)

Lines 421-423 are particularly interesting since they mention errors in map.qml, which is what drives the map window. 

```
2020-11-29 19:56:20.112 (W) qrc:/map/map.qml:159:49: Unexpected token `>'
2020-11-29 19:56:20.112 (W) qrc:/map/map.qml:160:36: Expected token `:'
2020-11-29 19:56:20.112 (W) qrc:/map/map.qml:160:47: Expected token `}'
```

My sdrangel-docker build is basically a stock version of 6.2.0 sdrangel-vanilla created with the provided scripts, with the only changes being a different timezone in the Dockerfile and mapping ~/.local from my machine to the container in run.sh (so that the databases are retained).

Thanks!

**Comment by spsx** [comment]
I modified the Dockerfile and managed to create an Ubuntu 20.10 Docker image, which has QT 5.14.2. The map works great as shown in the screenshot, however I also get graphical glitches when I move the GUI panes around (probably due to #21). So this isn't a viable solution.

![Screenshot from 2020-11-30 12-37-33](https://user-images.githubusercontent.com/19350831/100645346-8a6cae80-330a-11eb-9b01-cdf4ccd6f42f.png)

The map.qml errors have disappeared from the log.

[sdrangel_ubuntu20.log](https://github.com/f4exb/sdrangel-docker/files/5617718/sdrangel_ubuntu20.log)

Debian 10 doesn't have backports for a newer version of qt5, but I wonder if a manual install of a newer version is a viable solution...

**Comment by f4exb** [comment]
Thanks for the update. I had taken Debian 10 as a fallback from Ubuntu 20.04 that wasn't working for me in GUI mode and at that time Ubuntu 20.10 was not out yet. Now I have moved my dev environment to 20.10 as 20.04 even on bare metal has serious issues at least for me (Airspy HF support and Channel Analyzer destruction issue).

I can indeed consider moving the base image to 20.10

**Comment by spsx** [comment]
Yeah, the GUI is definitely glitchy with the 20.10 image, though the ADS-B functionality works great. If I have time I may look into running a newer version of QT on the Debian 10 image.

Thanks for your great work on sdrangel, with all the new demodulators sdrangel is becoming an all-in-one solution for SDR reception, rather than the current mess of piping and using different programs for different tasks :)

**Comment by spsx** [comment]
I set up a Debian 11 (Bullseye) Docker image, and it actually seems to work very well on sdrangel 6.2.0. Debian 11 uses QT 5.15.1. The ADS-B map shows up properly, and the GUI has no glitches as well from my testing. Digital and FM demod also works well on that image, with no visible issues.

Unfortunately there are some compile errors with the "soapysdr" device, therefore I ended up only enabling my rtl-sdr device in the Dockerfile. Likely there are some dependencies that I've missed; I had to change some parts of the Dockerfile to make things work.

Currently I'll be sticking with the Debian 11 image for my personal use of sdrangel. Since it's considered an "unstable" branch of Debian, I'm not sure if it's the best idea to roll it out to all users.
